packages/changed-elements-react/src/widgets/comparisonJobWidget/VersionCompareSelectDialogV2.tsx

```diff
@@ -0,0 +1,117 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { IModelApp, type IModelConnection } from "@itwin/core-frontend";
+import { Button, Modal, ModalButtonBar, ModalContent } from "@itwin/itwinui-react";
+import { useState, type ReactNode } from "react";
+
+import { VersionCompareUtils, VersionCompareVerboseMessages } from "../../api/VerboseMessages";
+import type { NamedVersion } from "../../clients/iModelsClient";
+import { useVersionCompare } from "../../VersionCompareContext";
+import { useNamedVersionLoader } from "./useNamedVersionLoader.js";
+import { VersionCompareSelectComponent } from "./VersionCompareSelectComponent";
+
+import "./VersionCompareSelectDialogV2.scss";
+
+/** Options for VersionCompareSelectDialogV2. */
+interface VersionCompareSelectDialogV2Props {
+  /** IModel Connection that is being visualized. */
+  iModelConnection: IModelConnection;
+
+  /** onClose triggered when user clicks start comparison or closes dialog. */
+  onClose: (() => void) | undefined;
+
+  "data-testid"?: string;
+
+  /** Optional prop for a user supplied component to handle managing named versions. */
+  manageNamedVersionsSlot?: ReactNode | undefined;
+}
+
+/**
+ * VersionCompareSelectDialogV2 use comparison jobs for processing. Requires context of:
+ * <VersionCompareContext iModelsClient={iModelsClient} comparisonJobClient={comparisonJobClient}>
+ *   ...
+ * </VersionCompareContext>
+ * ------------------------------------------------------------------------------------------------
+ * Should be used with provider. Example:
+ * <V2DialogProvider>
+ *   {
+ *     (isOpenCondition) &&
+ *     <VersionCompareSelectDialogV2
+ *       iModelConnection={this.props.iModelConnection}
+ *       onClose={this._handleVersionSelectDialogClose}
+ *     />
+ *   }
+ * </V2DialogProvider>
+ *
+ * Provider should be supplied with new dialog based on condition in order to keep
+ * track of toast and polling information.
+ *
+ * @throws Exception if context does not include iModelsClient and comparisonJobClient.
+*/
+export function VersionCompareSelectDialogV2(props: VersionCompareSelectDialogV2Props) {
+  const { comparisonJobClient } = useVersionCompare();
+  if (!comparisonJobClient) {
+    throw new Error("V2 Client is not initialized in given context.");
+  }
+
+  const { isLoading, result, prepareComparison } = useNamedVersionLoader(props.iModelConnection);
+
+  const [targetVersion, setTargetVersion] = useState<NamedVersion>();
+  const [currentVersion, setCurrentVersion] = useState<NamedVersion>();
+
+  const _handleOk = async (): Promise<void> => {
+    if (!comparisonJobClient || !result || !targetVersion || !currentVersion) {
```

Comment:
Will there be a case where comparisonJobClient is false here if checked in line 55

Reply:
Thanks for noticing this, 1accbd268f40c7d85ac803ba8a72973090235137 removes excessive check because we already do it within `useNamedVersionLoader`.